### DISCUSSION TITLE
Detect if editor is in diff view for context menus

### DISCUSSION THREAD

**Discussion by timheuer** [microsoft/vscode-discussions#678] [Extension Development QnA] [answered]
The thread at https://github.com/microsoft/vscode/issues/15513 is blocked alleging that this is solved via some ‘new API’ but doesn’t confirm what that is. The suggestions make it seem that it is a tabs api. But that doesn’t seem to allow for using `when` conditions in package.json

I have menus that show in `editor/title` but shouldn’t show when in a diff view. 

Can this be done with some activeviewid?

**Comment by isidorn** [comment, marked as answer]
Hey Tim. 
There seems to be the `isInDiffEditor` when context. So that should help you here. Here's the list of other when contexts:
https://code.visualstudio.com/api/references/when-clause-contexts#available-context-keys
